services/wallet/token/token.go

```diff
@@ -186,6 +191,21 @@ func (tm *Manager) FindToken(network *params.Network, tokenSymbol string) *Token
 	return nil
 }
 
+func (tm *Manager) LookupTokenIdentity(chainID uint64, address common.Address, native bool) *Token {
+	network := tm.networkManager.Find(chainID)
+	if native {
+		return tm.ToToken(network)
+	}
+
+	allTokens := tm.getFullTokenList(chainID)
```

Comment:
if I'm not missing something, this second part is identical to `FindTokenByAddress`

Reply:
😱 :) Thanks for the heads up